scripts/test/Muon/PlottingView_test.py

```diff
@@ -1,8 +1,445 @@
+                 self.mock_args,
+                 self.mock_kwargs)]}
+        self.view._replay_additions(self.plot_name)
+        args = [self.view, self.mock_name]
+        args.extend(self.mock_args)
+        self.mock_func.assert_called_once_with(*args)
+
+    def test_set_positions(self):
+        self.view._set_positions([[0, 0]])
+        self.plots_return_value.set_position.assert_called_once_with(
+            self.mock_grid_pos.get_position())
+        self.plots_return_value.set_subplotspec.assert_called_once_with(
+            self.mock_grid_pos)
+
+    def _set_update_plot_selector_and_set_positions_mocked(self):
+        self.view._set_positions = mock.Mock()
+        self.view._update_plot_selector = mock.Mock()
+
+    @mock.patch("Muon.GUI.ElementalAnalysis.Plotting.plotting_utils.get_layout")
+    def test_update_gridspec_if_new_plots(self, get_layout):
+        self._set_update_plot_selector_and_set_positions_mocked()
+        new_plots = [self.mock_plot]
+        self.view._update_gridspec(new_plots)
+        get_layout.assert_called_once_with(new_plots)
+        self.view._set_positions.assert_called_once_with(get_layout())
+        self.assertEquals(self.view._update_plot_selector.call_count, 1)
+
+    @mock.patch("Muon.GUI.ElementalAnalysis.Plotting.plotting_utils.get_layout")
+    def test_update_gridspec_if_new_plots_and_last(self, get_layout):
+        self._set_update_plot_selector_and_set_positions_mocked()
+        get_layout.return_value = [[0, 0]]
+        self.view._update_gridspec([self.mock_plot], last=self.plot_name)
+        self.view.figure.add_subplot.assert_called_once_with(
+            self.mock_grid_pos, label=self.plot_name)
+        self.plots_return_value.set_subplotspec.assert_called_once_with(
+            self.mock_grid_pos)
+        self.assertEquals(self.view._update_plot_selector.call_count, 1)
+
+    def test_gridspec_if_not_new_plots(self):
+        self.view._update_plot_selector = mock.Mock()
+        self.view._update_gridspec([])
+        self.assertEquals(self.view._update_plot_selector.call_count, 1)
+
+    def test_update_plot_selector(self):
+        self.view._update_plot_selector()
+        self.assertEquals(self.view.plot_selector.clear.call_count, 1)
+        self.view.plot_selector.addItems.assert_called_once_with(
+            list(self.view.plots.keys()))
+
+    def test_add_workspace_name_if_not_in_workspaces(self):
+        self.view._add_workspace_name(self.plot_name, self.mock_workspace)
+        self.assertEquals(self.view.workspaces[self.plot_name], [
+                          self.plots_return_value, self.mock_workspace])
+
+    def test_add_workspace_name_if_in_workspaces(self):
+        self.view.workspaces = {}
+        self.view._add_workspace_name(self.plot_name, self.mock_workspace)
+        self.assertEquals(self.view.workspaces[self.plot_name], [
+                          self.mock_workspace])
+
+
+class PlottingViewPlotFunctionsTests(unittest.TestCase):
```

Comment:
whats the difference between this and the class above?

Reply:
There are two reasons I remember: 
- To split them up by functionality
- The second class has a different set of methods and variables that need to be mocked that don't necessarily overlap with the first (and would mean having to unmock functions when called)

Reply:
could we have it in its own file? It is rather long with the two of them

Reply:
Can do

Reply:
see issue #23498